tests/unittest_brain_numpy_core_multiarray.py

```diff
@@ -77,105 +77,137 @@ def _inferred_numpy_func_call(self, func_name, *func_args):
         )
         return node.infer()
 
+    def _inferred_numpy_no_alias_func_call(self, func_name, *func_args):
+        node = builder.extract_node(
+            """
+        import numpy
+        func = numpy.{:s}
+        func({:s})
+        """.format(
+                func_name, ",".join(func_args)
+            )
+        )
+        return node.infer()
+
     def test_numpy_function_calls_inferred_as_ndarray(self):
         """
         Test that calls to numpy functions are inferred as numpy.ndarray
         """
-        for func_ in self.numpy_functions_returning_array:
-            with self.subTest(typ=func_):
-                inferred_values = list(self._inferred_numpy_func_call(*func_))
-                self.assertTrue(
-                    len(inferred_values) == 1,
-                    msg="Too much inferred values ({}) for {:s}".format(
-                        inferred_values, func_[0]
-                    ),
-                )
-                self.assertTrue(
-                    inferred_values[-1].pytype() == ".ndarray",
-                    msg="Illicit type for {:s} ({})".format(
-                        func_[0], inferred_values[-1].pytype()
-                    ),
-                )
+        for infer_wrapper in (
+            self._inferred_numpy_func_call,
+            self._inferred_numpy_no_alias_func_call,
+        ):
+            for func_ in self.numpy_functions_returning_array:
+                with self.subTest(typ=func_):
+                    inferred_values = list(infer_wrapper(*func_))
+                    self.assertTrue(
+                        len(inferred_values) == 1,
+                        msg="Too much inferred values ({}) for {:s}".format(
+                            inferred_values, func_[0]
+                        ),
+                    )
+                    self.assertTrue(
+                        inferred_values[-1].pytype() == ".ndarray",
+                        msg="Illicit type for {:s} ({})".format(
+                            func_[0], inferred_values[-1].pytype()
+                        ),
+                    )
 
     def test_numpy_function_calls_inferred_as_bool(self):
```

Comment:
```suggestion
INFER_WRAPPERS = [self._inferred_numpy_func_call, self._inferred_numpy_no_alias_func_call]

    @pytest.parametrize("infer_wrapper", INFER_WRAPPERS))
    def test_numpy_function_calls_inferred_as_bool(self, infer_wrapper ):
```
I think using parametrize would make the diff and the code more terse.

Reply:
Usually I would agree. However this test module still uses `unittest.TestCase` and in particular `self.subTest` which doesn't exist in `pytest` (without an extension). So using `@pytest.mark.parametrize` would require a complete rewrite which probably isn't worth it at this point.

Reply:
@Pierre-Sassoulas i didn't know about this `pytest` functionality. Thanks! I will remember it for my next devs.
@cdce8p thanks for your remark. I will let this as it is.